subset/dot1x/authenticator/authenticator.py

```diff
@@ -85,15 +85,19 @@ def __init__(self, config_file):
         self._radius_secret = None
         self._radius_id = None
         self._interface = None
+        self._enabled = None
 
         self._setup()
 
-    def _load_yaml_config(self):
+    def _load_config(self):
+        self.logger.info(self._config_file)
         with open(self._config_file, 'r') as file_stream:
-            config = yaml.safe_load(file_stream)
+            full_config = json.load(file_stream)
+        config = full_config.get('modules').get('dot1x')
 
         self.logger.debug('Loaded config from %s:\n %s', self._config_file, config)
 
+        self._enabled = config['enabled']
```

Comment:
this is likely overly fragile because it will break if the key doesn't exist. Usually I'd code this as ".get('enabled')" -- unless you specifically want it to break if the field is not defined... ?

Reply:
This test module won't even run when it's not enabled right? so there is no need to check for "enabled" inside the module.